types/listr/index.d.ts

```diff
@@ -0,0 +1,49 @@
+// Type definitions for listr 0.13
+// Project: https://github.com/samverschueren/listr#readme
+// Definitions by: Dusan Radovanovic <https://github.com/durad>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import * as stream from "stream";
+
+interface ListrRenderer {
+    nonTTY: boolean;
+    render(): void;
+    end(): void;
+}
+
+interface ListrOptions {
+    concurrent?: boolean | number;
+    exitOnError?: boolean;
+    renderer?: "silent" | "default" | "verbose" | ListrRenderer;
+    nonTTYRenderer?: "silent" | "default" | "verbose" | ListrRenderer;
+}
+
+interface ListrTask {
+    title: string;
+    output?: string;
+    task: (ctx: any, task: ListrTaskWrapper) => void | string | Promise<any> | stream.Readable | Listr;
+    skip?: (ctx: any, task: ListrTaskWrapper) => boolean | Promise<boolean> | string | void;
+    enabled?: (ctx: any, task: ListrTaskWrapper) => boolean | Promise<boolean>;
+}
+
+interface ListrTaskWrapper {
+    title: string;
+    output: any;
+    report(error: Error): void;
+    skip(message: string): void;
+    run(ctx?: any): Promise<any>;
+}
+
+declare class Listr {
+    constructor(tasks?: ReadonlyArray<ListrTask>, options?: ListrOptions);
+    constructor(options?: ListrOptions);
+    add(tasks: ListrTask | ReadonlyArray<ListrTask>): void;
+    run(ctx?: any): Promise<any>;
+}
+
+declare namespace Listr {
+}
```

Comment:
It seems the empty namespace can be removed

Reply:
Not really, if I remove empty namespace I get the following error when running `tsc`:

```
listr-tests.ts(1,24): error TS2497: Module '"DefinitelyTyped/types/listr/index"' 
resolves to a non-module entity and cannot be imported using this construct.
```

Reply:
You can use `import Listr = require("listr");` instead.
the namespance is a hack and should be avoided

Reply:
I've exported other sub-types through namespace. This way they can be used as types in code.